classes/module/Module.php

```diff
@@ -2451,6 +2451,9 @@ public function fetch($templatePath, $cache_id = null, $compile_id = null)
         if ($cache_id !== null) {
             Tools::enableCache();
         }
+        if ($compile_id == null) {
+            $compile_id = Context::getContext()->shop->theme->getName();
```

Comment:
I think the `compile_id` can be a mix between lang & shop & theme name because of smarty documentation:
>compile_id is an optional parameter. You can also set the $compile_id variable once instead of passing this to each call to this function. It is used in the event that you want to compile different versions of the same template, such as having separate templates compiled for different languages.

wdyt?

Reply:
Ok for the if conditions.
We althought concatain compileID

$compile_id .= Context::getContext()->shop->theme->getName();


I think it’s not necessary and not recommandes to add « dynamique » parameters. Compile ID is not a cache ID. Compilation is done to avoid parsing templates (transforme tpl into php) but keeping dynamic content. In some case like here, we have the  exact same relative path between two templates that’s explain the bug. We need to specify to smarty that we have several versions of template to compile.
So your suggestion to add id works but is not efficient.

If you supply to smarty dynamic compile Id like shop id, lang id, ... and (why not product id) you will multiply part of compile templates exponentially. On our case a Prestashop with 200 shops (yes it is possible) and 2 thèmes will generate too many unnecessary compiled files.

If a developper want to use cache with smarty (with cache id parameter), he need to supply obviously all id parameters.

So my fix doesn’t change any thing for those who defined on there module a  compile id because of the if null condition and consider they need a specific.

Reply:
Seems legit, thanks for explanation :+1: